tests/conftest.py

```diff
@@ -218,5 +219,27 @@ def trained_async(tmpdir_factory):
 def get_test_client(server):
     test_client = server.test_client
     test_client.port = None
-
     return test_client
+
+
+@contextmanager
+def log_emitted(
```

Comment:
Can you add a short docstring? I had to read the code before I understood what it actually did. Maybe also calling it sth like `assert_log_was_emitted` would help.